### DISCUSSION TITLE
[Marketplace] `package.json` validation and accidental extension delete prevention

### DISCUSSION THREAD

**Discussion by prashantvc** [microsoft/vscode-discussions#403] [Announcements]
We deployed improvements to two of extension author scenarios:

1. **Publisher name validation** - Live now
Security is of paramount importance for VS Marketplace consumers. The consumers expect the extensions in the VS Marketplace to be secure, reliable, and trusted. Currently the Marketplace does not validate if the `publisher` mentioned in the `package.json` matches with the uploader's.

2.  **Accidental extension delete prevention** - Will deploy later this week.
Marketplace do not have a sufficient safeguard for extension deletion from the web gallery/CLI. The lack data recovery mechanism paired with hard delete, we often receive support email from unhappy extension authors when they unintentionally remove an extension from the Marketplace.
We have added the additional confirmation UI when you're deleting an extension. If your extension is old and has few hundred download you will still need to contact to delete an extension.

While these are first steps in improving the extension author's experience, we are continuously improve the service and appreciate you feedback

**Comment by alefragnani** [comment]
Hi,

First of all, that’s great to see the improvements in the marketplace area. Congrats!

About the scenarios you provided…

### Publisher name validation

What a coincidence. We were discussing something related to this yesterday here https://vscode-dev-community.slack.com/archives/C74CB59NE/p1673358662096609, based on a post in https://blog.aquasec.com/can-you-trust-your-vscode-extensions. The recent efforts that has been delivered about publisher and repository signing/verification helps a lot, and any other improvements you can add to make the marketplace more secure and trusted, is welcome, not only for end users but for extension authors as well.

### Accidental extension delete prevention

From time to time I see extension authors complaining about it, not only because they did delete the extension unintentionally but also for not being able to unpublish some specific version (for whatever reason). 

IMHO, delete (wipe) an extension should be the _exception_ path, with as much steps/validations/parameters as possible. Maybe even ask for the PAT and/or full extension id (like deleting a repo in GitHub) to _be sure_ about what is going to happen.

Also, maybe the `unpublish` command should be changed to unpublish _some specific version_ instead of delete the extension. 

Thank you

**Reply by prashantvc** [reply]
Thanks a lot for your feedback. The post confirms that we are on the right path to fix the security/trust issues with the marketplace. I wonder if the blogpost author has access to our backlog 😛

The current platform makes it extremely difficult to implement changes on the legacy infrasturcture as we rely on, as we modernise the infra you will start to see improvements to publish/unpublish flow. And, ofcourse, we are currently working on unpublish specific version
